src/generate.ts

```diff
@@ -111,32 +111,53 @@ export async function generatePdfFiles(
     const sideBars: UnprocessedSidebars = loadSidebars(versionInfo.sidebarFile, sidebarOptions);
     console.log(`${pluginLogPrefix}Sidebar file '${versionInfo.sidebarFile}' loaded.`);
 
-    slugger = new GithubSlugger(); // forget slugs from other versions   
+    // Loop through all configured sidebar names
+    for (const [i, sidebarName] of pluginOptions.sidebarNames.entries()) {
 
-    // Create build folder for that version
-    const versionBuildDir = join(pdfBuildDir, versionInfo.urlAddIn);
-    fs.ensureDirSync(versionBuildDir);
+      slugger = new GithubSlugger();
+      let folderName = '';
+      let productTitle = '';
 
-    // Build URL to root document of that version
-    let rootDocUrl = `${siteAddress}docs/`
-    if (versionInfo.urlAddIn) {
-      rootDocUrl = `${rootDocUrl}${versionInfo.urlAddIn}/`;
-    }
+      if (pluginOptions.productTitles && pluginOptions.productTitles.length >= i) {
+        productTitle = pluginOptions.productTitles[i];
+      }
 
-    // Get root document id of that version (the markdown with slug set to '/')
-    let rootDocId = '';
-    for (const entry of pluginOptions.rootDocIds) {
-      if (entry.version === versionInfo.version) {
-        rootDocId = entry.rootDocId;
-        break;
+      if (pluginOptions.subfolders && pluginOptions.subfolders.length >= i) {
+        folderName = pluginOptions.subfolders[i];
       }
-      if (entry.version === 'default') {
-        rootDocId = entry.rootDocId;
+
+      // Create build folder for that version
+      const versionBuildDir = join(pdfBuildDir, versionInfo.urlAddIn, folderName);
+      fs.ensureDirSync(versionBuildDir);
+
+      // Build URL to root document of that version
+
+      let rootDocUrl = `${siteAddress}docs/`;
+      
+      if (folderName) {
+        rootDocUrl = `${siteAddress}docs/${folderName}/`;
+      }
+      
+      if (versionInfo.urlAddIn) {
+        rootDocUrl = `${rootDocUrl}${versionInfo.urlAddIn}/`;
       }
-    }
 
-    // Loop through all configured sidebar names
-    for (const sidebarName of pluginOptions.sidebarNames) {
+      
+      if (versionInfo.urlAddIn && versionInfo.urlAddIn) {
```

Comment:
This double check for same field seems to be a mistake.
Should there be another 2nd condition?

Reply:
Fixed the if statement. Now checks for versionInfo.urlAddIn and folderName